### PR TITLE
Service AnsibleTower and EmbeddedAnsible UI parity

### PR DESCRIPTION
**Pull request by tumido**
Make the service layout same for AnsibleTower as it is for EmbeddedAnsible - provider part

- Extract `:raw_stdout_via_worker` to shared Ansible codebase

Related PRs: 
Fixes: [BUG 1590840](https://bugzilla.redhat.com/show_bug.cgi?id=1590840) [BUG 1590844](https://bugzilla.redhat.com/show_bug.cgi?id=1590844)

### CODE REVIEW

**Comment by tumido** [comment]
@miq-bot add_label enhancement

**Comment by gmcculloug** [comment]
@jameswnl Is this something you are actively reviewing?  Should anyone else be included on the review list?

File: spec/models/manageiq/providers/ansible_tower/automation_manager/job_spec.rb [hunk 1]
```diff
@@ -1,3 +1,36 @@
 describe ManageIQ::Providers::AnsibleTower::AutomationManager::Job do
+  let(:job) { FactoryGirl.create(:ansible_tower_job) }
+
   it_behaves_like 'ansible job'
+
+  describe '#raw_stdout_via_worker' do
```
Comment (bzwei) [inline spec/models/manageiq/providers/ansible_tower/automation_manager/job_spec.rb]: can you move new spec code inside `it_hehaves_like 'ansible job'`?

File: app/models/manageiq/providers/ansible_tower/shared/automation_manager/job.rb [hunk 2]
```diff
@@ -168,6 +168,22 @@ def raw_stdout(format = 'txt')
     raise MiqException::MiqOrchestrationStatusError, err.to_s, err.backtrace
   end
 
+  # Intend to be called by UI to display stdout. The stdout is stored in MiqTask#task_results or #message if error
+  # Since the task_results may contain a large block of data, it is desired to remove the task upon receiving the data
+  def raw_stdout_via_worker(userid, format = 'txt', role = nil)
```
Comment (jameswnl) [inline app/models/manageiq/providers/ansible_tower/shared/automation_manager/job.rb]: This is putting on queue. So better name it something like `raw_stdout_queue` (like in [here](https://github.com/jameswnl/manageiq-providers-ansible_tower/blob/9cc3598ee1affc402f4ded2dcc6e2988ecd00064/app/models/manageiq/providers/ansible_tower/shared/automation_manager/tower_api.rb#L95-L95))

File: spec/models/manageiq/providers/ansible_tower/automation_manager/job_spec.rb (hunk 1 above)
Comment (tumido) [reply spec/models/manageiq/providers/ansible_tower/automation_manager/job_spec.rb]: It won't work since this feature is also used in Embedded Ansible and in that case the task requires `embedded_ansible` role to be provided by the server. Embedded Ansible has its implementation adjusted in this PR https://github.com/ManageIQ/manageiq/pull/17712 and their very own specs are available [here](https://github.com/ManageIQ/manageiq/blob/master/spec/models/manageiq/providers/embedded_ansible/automation_manager/job_spec.rb)

In other words: If moved to the shared example these specs would fail for Embedded Ansible.

Comment (jameswnl) [reply spec/models/manageiq/providers/ansible_tower/automation_manager/job_spec.rb]: @tumido it's likely doable to share the spec.  Try move [this line](https://github.com/ManageIQ/manageiq/blob/master/spec/models/manageiq/providers/embedded_ansible/automation_manager/job_spec.rb#L15) to L3 and the shared specs will receive it.

Comment (tumido) [reply spec/models/manageiq/providers/ansible_tower/automation_manager/job_spec.rb]: @jameswnl :+1: I've moved it. Let me know if the current spec layout is acceptable. :wink: I needed to change the context flow for Embedded Ansible job_spec. Please review the core PR as well. https://github.com/ManageIQ/manageiq/pull/17712

File: app/models/manageiq/providers/ansible_tower/shared/automation_manager/job.rb (hunk 2 above)
Comment (tumido) [reply app/models/manageiq/providers/ansible_tower/shared/automation_manager/job.rb]: @jameswnl ^ I can do it, but I would change it for both. What does @bzwei think?

**Comment by jameswnl** [comment]
@tumido 👍  thanks!
